src/Take.Blip.Builder/Variables/BaseResourceVariableProvider.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Lime.Protocol;
+using Lime.Protocol.Network;
+using Lime.Protocol.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Client;
+
+namespace Take.Blip.Builder.Variables
+{
+    public class BaseResourceVariableProvider : IVariableProvider
```

Comment:
Rename to `ResourceVariableProviderBase` to keep the same pattern used on this repo and make it `abstract`.